utils/mgmt_system/ec2.py

```diff
@@ -125,6 +127,39 @@ def delete_vm(self, instance_id):
         except ActionTimedOutError:
             return False
 
+    def describe_stack(self, stack_name):
+        """Describe stackapi
+
+        Returns the description for the specified stack
+        Args:
+            stack_name: Unique name of stack
+        """
+        result = []
+        stacks = self.stackapi.describe_stacks(stack_name)
+        result.extend(stacks)
+        return result
+
+    def stack_exist(self, stack_name):
+        stacks = [stack for stack in self.describe_stack(stack_name)
+            if stack.stack_name == stack_name]
+        if (len(stacks) >= 1):
```

Comment:
Sorry, missed also this one:

```python
if len(stacks) > 0:
```

but it can be just
```python
if stacks:
```

but even shorter

```python
return bool(stacks)
```

Actually the whole function could be written like this:

```python
def stack_exists(self, stack_name):
   return any(stack.stack_name == stack_name for stack in self.describe_stack(stack_name))
```

Pick anything you like. :)